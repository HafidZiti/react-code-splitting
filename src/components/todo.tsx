import React from "react";

export interface Itodo {
  id: number;
  title: string;
  completed: boolean;
}

type props = {
  todo: Itodo;
};

const Todo: React.FC<props> = ({ todo }) => {
  const { id, title, completed } = todo;
  const h1 = <h1>{title}</h1>;
  const text = completed ? <strong>{h1}</strong> : h1;
  return <div data-testid={`todo-${id}`}>{text}</div>;
};

export default Todo;
