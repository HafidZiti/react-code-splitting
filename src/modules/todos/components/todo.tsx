import React from "react";

export interface TodoProps {
  id: number;
  title: string;
  completed: boolean;
}

type props = {
  todo: TodoProps;
};

export const Todo: React.FC<props> = ({ todo }) => {
  const { id, title, completed } = todo;
  const h1 = <h1>{title}</h1>;
  const text = completed ? <strong>{h1}</strong> : h1;
  return <div data-testid={`todo-${id}`}>{text}</div>;
};
