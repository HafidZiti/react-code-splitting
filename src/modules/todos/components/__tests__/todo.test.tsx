import React from "react";
import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Todo, TodoProps } from "../todo";

afterEach(() => {
  cleanup();
});

test("should render non-completed todo component", () => {
  const mockProps: TodoProps = { id: 1, title: "task-1", completed: false };
  render(<Todo todo={mockProps} />);
  const todoElement = screen.getByTestId("todo-1");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("task-1");
  expect(todoElement).not.toContainHTML("<strong>");
});

test("should render completed todo component", () => {
  const mockProps: TodoProps = { id: 2, title: "task-2", completed: true };
  render(<Todo todo={mockProps} />);
  const todoElement = screen.getByTestId("todo-2");
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent("task-2");
  expect(todoElement).toContainHTML("<strong><h1>task-2</h1></strong>");
});

test("matches snapshot", () => {
  const mockProps: TodoProps = { id: 2, title: "task-2", completed: true };
  const tree = renderer.create(<Todo todo={mockProps} />).toJSON();
  console.log(tree);
  expect(tree).toMatchSnapshot();
});
