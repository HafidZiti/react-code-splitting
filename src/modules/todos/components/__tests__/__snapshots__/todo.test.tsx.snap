// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-2"
>
  <strong>
    <h1>
      task-2
    </h1>
  </strong>
</div>
`;
